### PR TITLE
chore: bump Node types to match supported Node version

### PR DESCRIPTION
**Pull request by Simek**
Summary:
---------

Refs: #1072

This small PR bumps the Node types package (`@types/node`) to the `10.0.0` version to match the current [minimal supported version](https://github.com/react-native-community/cli/blob/master/packages/cli/package.json#L21) of Node required for the CLI.

Test Plan:
----------

Successful `yarn install` run!

### CODE REVIEW

**Comment by Simek** [comment]
According to the CLI error - according to TS types it looks like emitting Signal to `process` [requires two parameters](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/process.d.ts#L322), where both of them are signals. Only REPL `emit` seems to [accept one argument](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/repl.d.ts#L309) (but not all Signals).

File: packages/cli/src/commands/start/watchMode.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ function enableWatchMode(messageSocket: any) {
           process.exit();
           break;
         case 'z':
-          process.emit('SIGTSTP');
+          process.emit('SIGTSTP', 'SIGTSTP');
```
Comment (grabbou) [inline packages/cli/src/commands/start/watchMode.ts:42]: Curious why these two (same) arguments are required? What they're supposed to do?

**Review by grabbou — APPROVED** [review]
LGTM, just a comment to better understand one thing.

File: packages/cli/src/commands/start/watchMode.ts (hunk 1 above)
Comment (Simek) [reply packages/cli/src/commands/start/watchMode.ts:42]: It looks like process emit is based on the Events emitter https://nodejs.org/api/events.html#events_emitter_emit_eventname_args and the second argument is passed to the user defined listener/handler, but I'm not 100% sure about this. 😉

**Comment by Simek** [comment]
BTW. Refs. https://github.com/facebook/react-native-website/pull/2267

I think that minimum Node version bump to 12 should also be considered for the CLI in the near future.

File: packages/cli/src/commands/start/watchMode.ts (hunk 1 above)
Comment (grabbou) [reply packages/cli/src/commands/start/watchMode.ts:42]: I see. We could've used it then to pass some useful data but I don't think we have a use case for it right now.

**Pull request merged**